src/test/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleClassResolverTest.java

```diff
@@ -0,0 +1,23 @@
+package org.jvnet.hudson.test.junit.jupiter;
+
+import org.junit.jupiter.api.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+
+@WithJenkins
+class JenkinsRuleClassResolverTest {
+
+    static JenkinsRule rule;
```

Comment:
Rather than reflection would it be cleaner to explicitly register it do you think?
https://junit.org/junit5/docs/current/user-guide/#extensions-registration-programmatic

Reply:
I was trying to avoid having different solutions for the different contexts. 

`@WithJenkins` on the class or method gets you test-method parameter injection. Having a different approach for the class test context creates additional complexity for the consuming devs.